app/Http/Controllers/CityController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\City;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+
+class CityController extends Controller
+{
+    public function index()
+    {
+        return view('cities.index');
+    }
+
+    public function getAll()
+    {
+        $cities = City::all();
+        return response()->json(['cities' => $cities]);
+    }
+
+    public function store(Request $request): JsonResponse
+    {
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'required|string|max:255|unique:cities,name',
+            ]
+        );
+
+        if ($validator->fails()) {
+            return response()->json([
+                'error' => $validator->messages()
+            ], 400);
+        }
+
+        $city = City::create([
+            'name' => $request->input('name'),
+        ]);
+
+        return response()->json($city, 201);
+    }
+
+    public function update(Request $request, $id): JsonResponse
+    {
+        $city = City::find($id);
+
+        if (! $city) {
+            return response()->json(['error' => 'City not found'], 404);
+        }
+
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'required|string|max:255|unique:cities,name',
+            ]
+        );
+
+        if ($validator->fails()) {
+            return response()->json([
+                'error' => $validator->messages()
+            ], 400);
+        }
+
+        $city->name = $request->input('name');
+        $city->save();
+        return response()->json($city, 200);
```

Comment:
You could use Flugg\Responder and define a transformer to respond with the created city